src/main/java/echoquery/sql/QueryBuilder.java

```diff
@@ -84,17 +92,11 @@ public static QueryBuilder of(Intent intent) {
     } else {
       aggregationFunc = QueryUtil.functionCall(
           SlotUtil.getAggregateFunction(aggregate),
-          QueryUtil.nameReference(aggregationColumn));
+          new QualifiedNameReference(
+              QualifiedName.of(from.wherePrefix(1), aggregationColumn)));
```

Comment:
the caller of getPrefix doesnt know whether to ask for source table or destination table, thats the point! you ask for the prefix of a column not a table